lib/index.js

```diff
@@ -13,5 +13,46 @@ module.exports = (dirname, manifestExtras) => {
 
     const manifest = Manifest.concat(manifestExtras || []);
 
-    return Haute(dirname, manifest);
+    return (server, options, next) => {
+
+        const promise = function () {
+
+            return new Promise((resolve, reject) => {
+
+                Haute(dirname, manifest)(server, options, (err) => {
+
+                    if (err) {
+                        reject(err);
+                    }
+                    else {
+                        resolve();
+                    }
+                });
+            });
+        };
+
+        if (typeof options !== 'undefined') {
```

Comment:
This stuff is very hard for me to understand!  I think it can be simpler.  Also, I see `options` still being passed in all cases.